src/DFG.cpp

```diff
@@ -263,6 +387,223 @@ void DFG::combineForUnroll(list<string>* t_targetPattern){
+  }
+  // combine succeed DFGNodes with number <= t_mergeSize
+  for (auto& pair: opcodeNameMap) {
+      if (pair.second.size() > 1) {
+        combineSuccSame(&pair.second, t_mergeSize);
+        found = true;
+      }
+  }
+  return found;
+}
+
+// findOpSucc finds nodes br which means the nodes are from different paths, t_mergeSize decides the max number of nodes that can be combined
+void DFG::findOpSucc(const list<DFGNode*>* t_succList, const int t_mergeSize) {
+  // std::cout <<"[MMJ] findOpSucc is running "<< std::endl;
+  // list<DFGNode*>* uniqueNodes;
+  // list<DFGNode*> succUniqueNodes;
+  // uniqueNodes = combineSuccSame(t_succList, t_mergeSize);
+  // // after merge nodes after br, we merge succeed nodes of the merged nodes
+  // while (true){
+  //   for (DFGNode* dfgNode: *uniqueNodes){
+  //     for (DFGNode* succNode: *(dfgNode->getSuccNodes())){
+  //       if (succNode->getPathName() == dfgNode->getPathName()){
+  //         // find succNodes of uniqueNodes and succNodes must in the same path with uniqueNodes
+  //         succUniqueNodes.push_back(succNode);
+  //       }
+  //     }
+  //   }
+  //   uniqueNodes =  combineSuccSame(&succUniqueNodes, t_mergeSize);
+  //   succUniqueNodes.clear();
+  //   if (uniqueNodes == NULL){
+  //     // finish if no uniqueNodes to be combined
+  //     std::cout <<"[MMJ] break the  findOpSucc"<< std::endl;
+  //     break;
+  //   }
+  // }
+  return;
+}
+
+// dualIssue is used to identify 'br' with multiple output, it combines the ouput together with the same control edge.
+// Note that the control edge is only one bit width since there is noly one 'br' input.
+// Note that dualIssue must be done after other combine operation.
+void DFG::dualIssue(const int t_mergeSize, const int t_mergeCount){
+  // testBench: adpcmkernel's 'br-3phi'
+  // toBeMatchedDFGNodes is to store the DFG nodes after br
+  std::cout <<"[MMJ] dual Issue is running. "<<std::endl;
+  list<DFGNode*>* toBeMatchedDFGNodes = new list<DFGNode*>();
+  int countPartialMerge = 1;
+  for (DFGNode* dfgNode: nodes) {
+    if(countPartialMerge > t_mergeCount){
+      return;
+    }
+    // we support both br and switch
+    if ((dfgNode->isBranch() or dfgNode->isOpt("switch")) and !dfgNode->hasMerged()) {
+      bool found = false;
+      std::cout <<"[MMJ] dual Issue finds the br/switch "<< dfgNode->getID() << std::endl;
+      for (DFGNode* succNode: *(dfgNode->getSuccNodes())) {
+        if (!succNode->hasMerged() and !succNode->hasCombined()){
+          toBeMatchedDFGNodes->push_back(succNode); 
+        }
+      }  
+      // find nodes that are the same after br
+      found = findOpSameSucc(toBeMatchedDFGNodes, t_mergeSize);
+      countPartialMerge++;
+    }
+    // clear toBeMatchedDFGNodes and find the next br/switch
+    toBeMatchedDFGNodes->clear();
+  }
+}
+
+
+// findIFELPath finds succeed nodes after br with different path
+void DFG::findIFELPath(list<DFGNode*>* t_succList, const int t_mergeSize) {
```

Comment:
`findIFELPath` -> `findAndMergeIfElsePaths`? Am I correct? It looks not only `find` but also `merge`?